hybrasyl/Objects/User.cs

```diff
@@ -2305,586 +2145,103 @@ public override void RegenerateMp(double mp, Creature regenerator = null)
-                            var dmg = rand.Next(Convert.ToInt32(simple.Min), Convert.ToInt32(simple.Max));
-                            //these need to be set to integers as attributes. note to fix.
-                            target.Damage(dmg, OffensiveElement, damageType, this);
-                        }
-                        else
-                        {
-                            var formula = damage.Formula;
-                            var damageType = EnumUtil.ParseEnum(damage.Type.ToString(), (Enums.DamageType)castObject.Effects.Damage.Type);
-                            var parser = new FormulaParser(this, castObject, target);
-                            var dmg = parser.Eval(formula);
-                            if (dmg == 0) dmg = 1;
-                            target.Damage(dmg, OffensiveElement, damageType, this);
-                        }
-
-                        if (castObject.Effects.Animations.OnCast.Target == null) continue;
-                        var effectAnimation = new ServerPacketStructures.EffectAnimation()
-                        {
-                            SourceId = this.Id,
-                            Speed = (short)castObject.Effects.Animations.OnCast.Target.Speed,
-                            TargetId = target.Id,
-                            TargetAnimation = castObject.Effects.Animations.OnCast.Target.Id
-                        };
-                        Enqueue(effectAnimation.Packet());
-                        SendAnimation(effectAnimation.Packet());
-                    }
+                    // TODO: handle castables with scripting
+                    // DoStuff();
+                    continue;
                 }
-
-                Motion motion;
-
-                try
-                {
-                    motion = castObject.Effects.Animations.OnCast.Player.SingleOrDefault(x => x.Class.Contains((Class)Class));
+                if (castObject.Effects?.Damage != null) { 
+                    var damageOutput = NumberCruncher.CalculateDamage(castObject, tar, this);
+                    Logger.Info($"UseCastable: {Name} casting {castObject.Name} - target: {tar.Name} damage: {damageOutput}");
+                    tar.Damage(damageOutput.Amount, OffensiveElement, damageOutput.Type, this);
                 }
-                catch (InvalidOperationException)
+                // Note that we ignore castables with both damage and healing effects present - one or the other.
+                // A future improvement might be to allow more complex effects.
+                else if (castObject.Effects.Heal != null)
                 {
-                    motion = castObject.Effects.Animations.OnCast.Player.FirstOrDefault(x => x.Class.Contains((Class)Class));
-
-                    Logger.ErrorFormat("{1}: contains more than one motion for a class definition, using first one found!", castObject.Name);
+                    var healOutput = NumberCruncher.CalculateHeal(castObject, tar, this);

```

Comment:
Nothing appears to happen with healOutput after this value assignment.

Reply:
Fixed, pebkac